2024-05-17 00:36:14,676 INFO \u0418\u043c\u044f: \u0422\u0443\u0437\u0438\u043a \u0412\u0438\u0434: \u0421\u043e\u0431\u0430\u043a\u0430 \u0421\u0435\u043c\u0435\u0439\u0441\u0442\u0432\u043e: \u041f\u0441\u043e\u0432\u044b\u0435 \u0438\u0437\u043c\u0435\u043d\u0435\u043d.
2024-05-17 00:37:33,986 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 00:40:10,654 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 00:40:25,427 INFO Имя: Тузик Вид: Собака Семейство: Псовые изменен.
2024-05-17 00:40:53,732 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 00:41:02,743 INFO Frick Ronald вошел в аккаунт.
2024-05-17 00:41:07,927 INFO Frick Ronald вышел из аккаунта.
2024-05-17 00:42:08,957 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\ZooparkApp\settings.py changed, reloading.
2024-05-17 00:42:09,584 INFO Watching for file changes with StatReloader
2024-05-17 00:42:11,366 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 00:42:11,919 INFO Watching for file changes with StatReloader
2024-05-17 00:42:18,950 INFO Щур Марат вошел в аккаунт.
2024-05-17 00:44:24,105 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 00:44:24,674 INFO Watching for file changes with StatReloader
2024-05-17 00:44:40,115 ERROR Internal Server Error: /auth/logout/
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py", line 47, in logoutUser
    logging.info(f"{request.user.position} {request.user.last_name} {request.user.first_name} вышел из аккаунта.")
                ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\utils\functional.py", line 253, in inner
    return func(_wrapped, *args)
           ^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'User' object has no attribute 'position'
2024-05-17 00:44:53,461 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 00:44:54,101 INFO Watching for file changes with StatReloader
2024-05-17 00:44:55,167 INFO Щур Марат вышел из аккаунта.
2024-05-17 00:46:57,638 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 01:00:39,626 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\urls.py changed, reloading.
2024-05-17 01:00:41,305 INFO Watching for file changes with StatReloader
2024-05-17 01:01:24,215 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\urls.py changed, reloading.
2024-05-17 01:01:24,749 INFO Watching for file changes with StatReloader
2024-05-17 01:04:31,725 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 01:04:32,284 INFO Watching for file changes with StatReloader
2024-05-17 01:04:51,270 INFO Watching for file changes with StatReloader
2024-05-17 01:05:04,447 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\urls.py changed, reloading.
2024-05-17 01:05:04,978 INFO Watching for file changes with StatReloader
2024-05-17 01:09:26,539 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\urls.py changed, reloading.
2024-05-17 01:09:28,204 INFO Watching for file changes with StatReloader
2024-05-17 01:13:17,918 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 01:13:18,483 INFO Watching for file changes with StatReloader
2024-05-17 01:13:40,655 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 01:13:41,291 INFO Watching for file changes with StatReloader
2024-05-17 01:14:15,156 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 01:14:15,759 INFO Watching for file changes with StatReloader
2024-05-17 01:14:26,996 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 01:14:27,575 INFO Watching for file changes with StatReloader
2024-05-17 01:19:40,760 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\services\cat_facts.py changed, reloading.
2024-05-17 01:19:41,378 INFO Watching for file changes with StatReloader
2024-05-17 01:19:59,779 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\services\dog_photos.py changed, reloading.
2024-05-17 01:20:00,412 INFO Watching for file changes with StatReloader
2024-05-17 01:20:03,703 INFO Получено случайное фото собаки.
2024-05-17 01:20:05,622 INFO Получен случайный факт про кошек.
2024-05-17 01:20:07,083 INFO Получено случайное фото собаки.
2024-05-17 01:20:08,660 INFO Получен случайный факт про кошек.
2024-05-17 01:20:09,420 INFO Получено случайное фото собаки.
2024-05-17 01:20:10,451 INFO Получен случайный факт про кошек.
2024-05-17 01:20:10,996 INFO Получено случайное фото собаки.
2024-05-17 01:20:11,826 INFO Получен случайный факт про кошек.
2024-05-17 01:20:13,321 INFO Получен случайный факт про кошек.
2024-05-17 01:20:13,644 INFO Получено случайное фото собаки.
2024-05-17 01:20:34,883 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 01:20:38,107 INFO Frick Ronald вошел в аккаунт.
2024-05-17 01:20:39,720 INFO Получено случайное фото собаки.
2024-05-17 01:20:41,773 INFO Получен случайный факт про кошек.
2024-05-17 01:20:42,739 INFO Frick Ronald вышел из аккаунта.
2024-05-17 01:21:05,933 WARNING Not Found: /cat-fact/
2024-05-17 01:21:19,090 INFO Frick Ronald вошел в аккаунт.
2024-05-17 01:21:21,444 INFO Получен случайный факт про кошек.
2024-05-17 01:21:32,424 INFO Frick Ronald вышел из аккаунта.
2024-05-17 01:21:42,787 INFO Получен случайный факт про кошек.
2024-05-17 01:22:52,360 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 01:22:52,992 INFO Watching for file changes with StatReloader
2024-05-17 01:23:07,367 INFO Frick Ronald вошел в аккаунт.
2024-05-17 01:23:09,127 INFO Получен случайный факт про кошек.
2024-05-17 01:23:12,470 INFO Frick Ronald вышел из аккаунта.
2024-05-17 01:23:19,431 WARNING Not Found: /accounts/login/
2024-05-17 01:23:37,492 INFO Frick Ronald вошел в аккаунт.
2024-05-17 01:23:40,511 INFO Получен случайный факт про кошек.
2024-05-17 01:23:42,475 INFO Получено случайное фото собаки.
2024-05-17 01:23:48,930 INFO Frick Ronald вышел из аккаунта.
2024-05-17 01:23:55,063 INFO Получено случайное фото собаки.
2024-05-17 01:24:55,502 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 01:24:56,115 INFO Watching for file changes with StatReloader
2024-05-17 01:25:03,079 WARNING Not Found: /accounts/login/
2024-05-17 01:25:11,240 INFO Frick Ronald вошел в аккаунт.
2024-05-17 01:25:18,784 INFO Получено случайное фото собаки.
2024-05-17 01:25:21,647 INFO Получено случайное фото собаки.
2024-05-17 01:25:24,451 INFO Получено случайное фото собаки.
2024-05-17 01:31:16,248 INFO Frick Ronald вышел из аккаунта.
2024-05-17 01:31:21,667 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 01:41:30,430 INFO Watching for file changes with StatReloader
2024-05-17 01:41:35,895 ERROR Internal Server Error: /
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 16, in index
    animals = Animal.objects.filter(name__icontains==name)
                                    ^^^^^^^^^^^^^^^
NameError: name 'name__icontains' is not defined
2024-05-17 01:41:47,976 ERROR Internal Server Error: /
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 16, in index
    animals = Animal.objects.filter(name__icontains==name)
                                    ^^^^^^^^^^^^^^^
NameError: name 'name__icontains' is not defined
2024-05-17 01:44:24,152 INFO Watching for file changes with StatReloader
2024-05-17 01:44:27,656 ERROR Internal Server Error: /
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 16, in index
    animals = Animal.objects.filter(name==name)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\query.py", line 1476, in filter
    return self._filter_or_exclude(False, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\query.py", line 1494, in _filter_or_exclude
    clone._filter_or_exclude_inplace(negate, args, kwargs)
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\query.py", line 1501, in _filter_or_exclude_inplace
    self._query.add_q(Q(*args, **kwargs))
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\sql\query.py", line 1613, in add_q
    clause, _ = self._add_q(q_object, self.used_aliases)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\sql\query.py", line 1645, in _add_q
    child_clause, needed_inner = self.build_filter(
                                 ^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\sql\query.py", line 1492, in build_filter
    arg, value = filter_expr
    ^^^^^^^^^^
TypeError: cannot unpack non-iterable bool object
2024-05-17 01:44:52,705 INFO Watching for file changes with StatReloader
2024-05-17 01:45:23,148 ERROR Internal Server Error: /
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 16, in index
    animals = Animal.objects.filter(name==name)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\query.py", line 1476, in filter
    return self._filter_or_exclude(False, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\query.py", line 1494, in _filter_or_exclude
    clone._filter_or_exclude_inplace(negate, args, kwargs)
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\query.py", line 1501, in _filter_or_exclude_inplace
    self._query.add_q(Q(*args, **kwargs))
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\sql\query.py", line 1613, in add_q
    clause, _ = self._add_q(q_object, self.used_aliases)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\sql\query.py", line 1645, in _add_q
    child_clause, needed_inner = self.build_filter(
                                 ^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\sql\query.py", line 1492, in build_filter
    arg, value = filter_expr
    ^^^^^^^^^^
TypeError: cannot unpack non-iterable bool object
2024-05-17 01:45:51,474 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 01:45:53,464 INFO Watching for file changes with StatReloader
2024-05-17 01:45:57,694 INFO Watching for file changes with StatReloader
2024-05-17 01:46:02,381 ERROR Internal Server Error: /
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 16, in index
    animals = Animal.objects.filter(name_icontains=name)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\query.py", line 1476, in filter
    return self._filter_or_exclude(False, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\query.py", line 1494, in _filter_or_exclude
    clone._filter_or_exclude_inplace(negate, args, kwargs)
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\query.py", line 1501, in _filter_or_exclude_inplace
    self._query.add_q(Q(*args, **kwargs))
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\sql\query.py", line 1613, in add_q
    clause, _ = self._add_q(q_object, self.used_aliases)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\sql\query.py", line 1645, in _add_q
    child_clause, needed_inner = self.build_filter(
                                 ^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\sql\query.py", line 1495, in build_filter
    lookups, parts, reffed_expression = self.solve_lookup_type(arg, summarize)
                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\sql\query.py", line 1307, in solve_lookup_type
    _, field, _, lookup_parts = self.names_to_path(lookup_splitted, self.get_meta())
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\sql\query.py", line 1772, in names_to_path
    raise FieldError(
django.core.exceptions.FieldError: Cannot resolve keyword 'name_icontains' into field. Choices are: aviary, aviary_id, daily_norm, date_of_birth, date_of_receipt, description, employee, employee_id, family, family_id, feed, feeding_schedule, id, image, name, type, type_id
2024-05-17 01:46:30,722 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 01:46:31,319 INFO Watching for file changes with StatReloader
2024-05-17 01:47:21,925 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 01:47:22,537 INFO Watching for file changes with StatReloader
2024-05-17 01:47:37,239 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 01:47:37,889 INFO Watching for file changes with StatReloader
2024-05-17 01:50:52,267 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 01:50:52,859 INFO Watching for file changes with StatReloader
2024-05-17 01:52:42,305 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 01:52:42,934 INFO Watching for file changes with StatReloader
2024-05-17 01:54:28,268 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 01:54:28,884 INFO Watching for file changes with StatReloader
2024-05-17 01:54:47,569 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 01:54:48,216 INFO Watching for file changes with StatReloader
2024-05-17 01:57:07,246 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 01:59:28,005 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 01:59:29,693 INFO Watching for file changes with StatReloader
2024-05-17 01:59:43,951 ERROR Internal Server Error: /
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 19, in index
    animals.order_by(age)
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\query.py", line 1701, in order_by
    obj.query.add_ordering(*field_names)
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\sql\query.py", line 2262, in add_ordering
    raise FieldError("Invalid order_by arguments: %s" % errors)
django.core.exceptions.FieldError: Invalid order_by arguments: [True]
2024-05-17 02:00:06,072 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 02:00:06,709 INFO Watching for file changes with StatReloader
2024-05-17 02:00:08,959 ERROR Internal Server Error: /
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 19, in index
    animals.order_by("age")
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\query.py", line 1701, in order_by
    obj.query.add_ordering(*field_names)
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\sql\query.py", line 2253, in add_ordering
    self.names_to_path(item.split(LOOKUP_SEP), self.model._meta)
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\sql\query.py", line 1772, in names_to_path
    raise FieldError(
django.core.exceptions.FieldError: Cannot resolve keyword 'age' into field. Choices are: aviary, aviary_id, daily_norm, date_of_birth, date_of_receipt, description, employee, employee_id, family, family_id, feed, feeding_schedule, id, image, name, type, type_id
2024-05-17 02:00:30,470 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 02:00:31,077 INFO Watching for file changes with StatReloader
2024-05-17 02:02:05,025 INFO Watching for file changes with StatReloader
2024-05-17 02:03:36,609 INFO Watching for file changes with StatReloader
2024-05-17 02:03:52,129 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 02:03:52,715 INFO Watching for file changes with StatReloader
2024-05-17 02:04:34,069 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 02:04:34,699 INFO Watching for file changes with StatReloader
2024-05-17 02:05:52,294 INFO Получен случайный факт про кошек.
2024-05-17 02:05:53,573 INFO Получено случайное фото собаки.
2024-05-17 02:14:07,559 ERROR Internal Server Error: /^add-animal/$
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 163, in add_animal
    logging.info(f"Имя: {animal.name} Вид: {animal.type} Семейство: {animal.family.name} добавлен в зоопарк.")
                         ^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'name'
2024-05-17 02:14:36,725 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 02:14:39,182 INFO Watching for file changes with StatReloader
2024-05-17 02:14:44,041 ERROR Internal Server Error: /^add-animal/$
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 163, in add_animal
    logging.info(f"Имя: {animal.name} Вид: {animal.type} Семейство: {animal.family} добавлен в зоопарк.")
                         ^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'name'
2024-05-17 02:15:15,748 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 02:15:17,121 INFO Watching for file changes with StatReloader
2024-05-17 02:15:21,655 INFO Имя: Майя Вид: 5 Семейство: 5 добавлен в зоопарк.
2024-05-17 02:17:03,336 ERROR Internal Server Error: /^delete-animal/(?P9\d+)/$
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 205, in delete_animal
    name = animal.name
           ^^^^^^^^^^^
AttributeError: 'QuerySet' object has no attribute 'name'
2024-05-17 02:17:47,290 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 02:17:48,717 INFO Watching for file changes with StatReloader
2024-05-17 02:17:52,333 INFO Имя: Майя Вид: Анаконда Семейство: Ложноногие удален из зоопарка.
2024-05-17 02:18:44,583 INFO Имя: Майя Вид: 5 Семейство: 5 добавлен в зоопарк.
2024-05-17 02:19:07,337 INFO Имя: Майя Вид: Анаконда Семейство: Ложноногие изменен.
2024-05-17 02:20:04,255 INFO Получен случайный факт про кошек.
2024-05-17 02:24:22,568 INFO Имя: Грейс Вид: 6 Семейство: 5 добавлен в зоопарк.
2024-05-17 02:24:25,027 ERROR Internal Server Error: /
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 875, in _resolve_lookup
    current = current[bit]
              ~~~~~~~^^^^^
TypeError: 'ImageFieldFile' object is not subscriptable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 21, in index
    return render(request, "zoopark/index.html", context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\test\utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\test\utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\defaulttags.py", line 242, in render
    nodelist.append(node.render_annotated(context))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\templatetags\static.py", line 116, in render
    url = self.url(context)
          ^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\templatetags\static.py", line 112, in url
    path = self.path.resolve(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 710, in resolve
    obj = self.var.resolve(context)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 842, in resolve
    value = self._resolve_lookup(context)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 885, in _resolve_lookup
    current = getattr(current, bit)
              ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\fields\files.py", line 66, in url
    self._require_file()
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\fields\files.py", line 41, in _require_file
    raise ValueError(
ValueError: The 'image' attribute has no file associated with it.
2024-05-17 02:25:43,047 ERROR Internal Server Error: /
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 875, in _resolve_lookup
    current = current[bit]
              ~~~~~~~^^^^^
TypeError: 'ImageFieldFile' object is not subscriptable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 21, in index
    return render(request, "zoopark/index.html", context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\test\utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\test\utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\defaulttags.py", line 242, in render
    nodelist.append(node.render_annotated(context))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\templatetags\static.py", line 116, in render
    url = self.url(context)
          ^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\templatetags\static.py", line 112, in url
    path = self.path.resolve(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 710, in resolve
    obj = self.var.resolve(context)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 842, in resolve
    value = self._resolve_lookup(context)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 885, in _resolve_lookup
    current = getattr(current, bit)
              ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\fields\files.py", line 66, in url
    self._require_file()
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\fields\files.py", line 41, in _require_file
    raise ValueError(
ValueError: The 'image' attribute has no file associated with it.
2024-05-17 02:31:43,136 INFO Имя: Майя Вид: Анаконда Семейство: Ложноногие изменен.
2024-05-17 02:32:34,447 ERROR Internal Server Error: /statistic/
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\backend_bases.py", line 2961, in _wait_cursor_for_draw_cm
    self.canvas.set_cursor(tools.Cursors.WAIT)
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\backends\_backend_tk.py", line 434, in set_cursor
    self._tkcanvas.configure(cursor=cursord[cursor])
  File "C:\Users\drine\AppData\Local\Programs\Python\Python312\Lib\tkinter\__init__.py", line 1721, in configure
    return self._configure('configure', cnf, kw)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\drine\AppData\Local\Programs\Python\Python312\Lib\tkinter\__init__.py", line 1711, in _configure
    self.tk.call(_flatten((self._w, cmd)) + self._options(cnf))
RuntimeError: main thread is not in main loop

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 107, in statistic
    get_graphic()
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 122, in get_graphic
    plt.savefig("zoopark/static/media/images/graph.png")
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\pyplot.py", line 1223, in savefig
    res = fig.savefig(*args, **kwargs)  # type: ignore[func-returns-value]
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\figure.py", line 3388, in savefig
    self.canvas.print_figure(fname, **kwargs)
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\backend_bases.py", line 2218, in print_figure
    result = print_method(
             ^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\backend_bases.py", line 2068, in <lambda>
    print_method = functools.wraps(meth)(lambda *args, **kwargs: meth(
                                                                 ^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\backends\backend_agg.py", line 496, in print_png
    self._print_pil(filename_or_obj, "png", pil_kwargs, metadata)
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\backends\backend_agg.py", line 444, in _print_pil
    FigureCanvasAgg.draw(self)
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\backends\backend_agg.py", line 385, in draw
    with (self.toolbar._wait_cursor_for_draw_cm() if self.toolbar
  File "C:\Users\drine\AppData\Local\Programs\Python\Python312\Lib\contextlib.py", line 137, in __enter__
    return next(self.gen)
           ^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\backend_bases.py", line 2964, in _wait_cursor_for_draw_cm
    self.canvas.set_cursor(self._last_cursor)
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\backends\_backend_tk.py", line 434, in set_cursor
    self._tkcanvas.configure(cursor=cursord[cursor])
  File "C:\Users\drine\AppData\Local\Programs\Python\Python312\Lib\tkinter\__init__.py", line 1721, in configure
    return self._configure('configure', cnf, kw)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\drine\AppData\Local\Programs\Python\Python312\Lib\tkinter\__init__.py", line 1711, in _configure
    self.tk.call(_flatten((self._w, cmd)) + self._options(cnf))
RuntimeError: main thread is not in main loop
2024-05-17 02:32:40,037 ERROR Internal Server Error: /statistic/
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\backend_bases.py", line 2961, in _wait_cursor_for_draw_cm
    self.canvas.set_cursor(tools.Cursors.WAIT)
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\backends\_backend_tk.py", line 434, in set_cursor
    self._tkcanvas.configure(cursor=cursord[cursor])
  File "C:\Users\drine\AppData\Local\Programs\Python\Python312\Lib\tkinter\__init__.py", line 1721, in configure
    return self._configure('configure', cnf, kw)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\drine\AppData\Local\Programs\Python\Python312\Lib\tkinter\__init__.py", line 1711, in _configure
    self.tk.call(_flatten((self._w, cmd)) + self._options(cnf))
RuntimeError: main thread is not in main loop

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 107, in statistic
    get_graphic()
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 122, in get_graphic
    plt.savefig("zoopark/static/media/images/graph.png")
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\pyplot.py", line 1223, in savefig
    res = fig.savefig(*args, **kwargs)  # type: ignore[func-returns-value]
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\figure.py", line 3388, in savefig
    self.canvas.print_figure(fname, **kwargs)
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\backend_bases.py", line 2218, in print_figure
    result = print_method(
             ^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\backend_bases.py", line 2068, in <lambda>
    print_method = functools.wraps(meth)(lambda *args, **kwargs: meth(
                                                                 ^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\backends\backend_agg.py", line 496, in print_png
    self._print_pil(filename_or_obj, "png", pil_kwargs, metadata)
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\backends\backend_agg.py", line 444, in _print_pil
    FigureCanvasAgg.draw(self)
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\backends\backend_agg.py", line 385, in draw
    with (self.toolbar._wait_cursor_for_draw_cm() if self.toolbar
  File "C:\Users\drine\AppData\Local\Programs\Python\Python312\Lib\contextlib.py", line 137, in __enter__
    return next(self.gen)
           ^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\backend_bases.py", line 2964, in _wait_cursor_for_draw_cm
    self.canvas.set_cursor(self._last_cursor)
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\backends\_backend_tk.py", line 434, in set_cursor
    self._tkcanvas.configure(cursor=cursord[cursor])
  File "C:\Users\drine\AppData\Local\Programs\Python\Python312\Lib\tkinter\__init__.py", line 1721, in configure
    return self._configure('configure', cnf, kw)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\drine\AppData\Local\Programs\Python\Python312\Lib\tkinter\__init__.py", line 1711, in _configure
    self.tk.call(_flatten((self._w, cmd)) + self._options(cnf))
RuntimeError: main thread is not in main loop
2024-05-17 02:32:55,926 INFO Watching for file changes with StatReloader
2024-05-17 02:33:15,073 ERROR Internal Server Error: /statistic/
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\backend_bases.py", line 2961, in _wait_cursor_for_draw_cm
    self.canvas.set_cursor(tools.Cursors.WAIT)
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\backends\_backend_tk.py", line 434, in set_cursor
    self._tkcanvas.configure(cursor=cursord[cursor])
  File "C:\Users\drine\AppData\Local\Programs\Python\Python312\Lib\tkinter\__init__.py", line 1721, in configure
    return self._configure('configure', cnf, kw)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\drine\AppData\Local\Programs\Python\Python312\Lib\tkinter\__init__.py", line 1711, in _configure
    self.tk.call(_flatten((self._w, cmd)) + self._options(cnf))
RuntimeError: main thread is not in main loop

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 107, in statistic
    get_graphic()
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 122, in get_graphic
    plt.savefig("zoopark/static/media/images/graph.png")
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\pyplot.py", line 1223, in savefig
    res = fig.savefig(*args, **kwargs)  # type: ignore[func-returns-value]
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\figure.py", line 3388, in savefig
    self.canvas.print_figure(fname, **kwargs)
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\backend_bases.py", line 2218, in print_figure
    result = print_method(
             ^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\backend_bases.py", line 2068, in <lambda>
    print_method = functools.wraps(meth)(lambda *args, **kwargs: meth(
                                                                 ^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\backends\backend_agg.py", line 496, in print_png
    self._print_pil(filename_or_obj, "png", pil_kwargs, metadata)
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\backends\backend_agg.py", line 444, in _print_pil
    FigureCanvasAgg.draw(self)
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\backends\backend_agg.py", line 385, in draw
    with (self.toolbar._wait_cursor_for_draw_cm() if self.toolbar
  File "C:\Users\drine\AppData\Local\Programs\Python\Python312\Lib\contextlib.py", line 137, in __enter__
    return next(self.gen)
           ^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\backend_bases.py", line 2964, in _wait_cursor_for_draw_cm
    self.canvas.set_cursor(self._last_cursor)
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\matplotlib\backends\_backend_tk.py", line 434, in set_cursor
    self._tkcanvas.configure(cursor=cursord[cursor])
  File "C:\Users\drine\AppData\Local\Programs\Python\Python312\Lib\tkinter\__init__.py", line 1721, in configure
    return self._configure('configure', cnf, kw)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\drine\AppData\Local\Programs\Python\Python312\Lib\tkinter\__init__.py", line 1711, in _configure
    self.tk.call(_flatten((self._w, cmd)) + self._options(cnf))
RuntimeError: main thread is not in main loop
2024-05-17 02:40:48,796 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 02:40:51,249 INFO Watching for file changes with StatReloader
2024-05-17 02:41:27,796 ERROR Internal Server Error: /statistic/
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 108, in statistic
    context["plot"] = get_graphic()
                  ^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 116, in get_graphic
    fig = go.Figure(data=[go.Bar(x=families, y=counts)])
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\plotly\graph_objs\_bar.py", line 3353, in __init__
    self["x"] = _v
    ~~~~^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\plotly\basedatatypes.py", line 4874, in __setitem__
    self._set_prop(prop, value)
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\plotly\basedatatypes.py", line 5218, in _set_prop
    raise err
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\plotly\basedatatypes.py", line 5213, in _set_prop
    val = validator.validate_coerce(val)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\_plotly_utils\basevalidators.py", line 412, in validate_coerce
    self.raise_invalid_val(v)
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\_plotly_utils\basevalidators.py", line 296, in raise_invalid_val
    raise ValueError(
ValueError: 
    Invalid value of type 'django.db.models.query.QuerySet' received for the 'x' property of bar
        Received value: <QuerySet ['Кошачьи', 'Псовые', 'Кенгуровые', 'Ложноногие']>

    The 'x' property is an array that may be specified as a tuple,
    list, numpy array, or pandas Series
2024-05-17 02:42:40,853 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 02:42:42,168 INFO Watching for file changes with StatReloader
2024-05-17 02:43:50,894 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 02:43:52,158 INFO Watching for file changes with StatReloader
2024-05-17 02:44:14,985 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 02:44:29,508 INFO London Jack вошел в аккаунт.
2024-05-17 02:44:34,667 INFO London Jack вышел из аккаунта.
2024-05-17 02:44:41,031 INFO Redkins Bill вошел в аккаунт.
2024-05-17 02:44:54,486 WARNING Forbidden (CSRF token from POST incorrect.): /admin/login/
2024-05-17 02:44:57,556 WARNING Forbidden (CSRF token from POST incorrect.): /admin/login/
2024-05-17 02:45:39,133 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 02:45:45,018 INFO London Jack вошел в аккаунт.
2024-05-17 02:46:31,699 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 02:46:39,894 INFO London Jack вошел в аккаунт.
2024-05-17 02:47:51,565 INFO London Jack вышел из аккаунта.
2024-05-17 02:48:00,211 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 02:51:24,598 INFO Имя: Марко Вид: 7 Семейство: 6 добавлен в зоопарк.
2024-05-17 02:51:31,184 ERROR Internal Server Error: /
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 875, in _resolve_lookup
    current = current[bit]
              ~~~~~~~^^^^^
TypeError: 'ImageFieldFile' object is not subscriptable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 21, in index
    return render(request, "zoopark/index.html", context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\test\utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\test\utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\defaulttags.py", line 242, in render
    nodelist.append(node.render_annotated(context))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\templatetags\static.py", line 116, in render
    url = self.url(context)
          ^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\templatetags\static.py", line 112, in url
    path = self.path.resolve(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 710, in resolve
    obj = self.var.resolve(context)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 842, in resolve
    value = self._resolve_lookup(context)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 885, in _resolve_lookup
    current = getattr(current, bit)
              ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\fields\files.py", line 66, in url
    self._require_file()
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\fields\files.py", line 41, in _require_file
    raise ValueError(
ValueError: The 'image' attribute has no file associated with it.
2024-05-17 02:55:02,799 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\forms.py changed, reloading.
2024-05-17 02:55:03,896 INFO Watching for file changes with StatReloader
2024-05-17 02:56:22,719 WARNING Not Found: /^edit-animal/$
2024-05-17 02:56:35,942 ERROR Internal Server Error: /
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 875, in _resolve_lookup
    current = current[bit]
              ~~~~~~~^^^^^
TypeError: 'ImageFieldFile' object is not subscriptable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 21, in index
    return render(request, "zoopark/index.html", context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\test\utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\test\utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\defaulttags.py", line 242, in render
    nodelist.append(node.render_annotated(context))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\templatetags\static.py", line 116, in render
    url = self.url(context)
          ^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\templatetags\static.py", line 112, in url
    path = self.path.resolve(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 710, in resolve
    obj = self.var.resolve(context)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 842, in resolve
    value = self._resolve_lookup(context)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 885, in _resolve_lookup
    current = getattr(current, bit)
              ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\fields\files.py", line 66, in url
    self._require_file()
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\fields\files.py", line 41, in _require_file
    raise ValueError(
ValueError: The 'image' attribute has no file associated with it.
2024-05-17 02:57:20,772 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\models.py changed, reloading.
2024-05-17 02:57:21,893 INFO Watching for file changes with StatReloader
2024-05-17 02:58:06,105 INFO Watching for file changes with StatReloader
2024-05-17 02:58:17,159 ERROR Internal Server Error: /
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 875, in _resolve_lookup
    current = current[bit]
              ~~~~~~~^^^^^
TypeError: 'ImageFieldFile' object is not subscriptable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 21, in index
    return render(request, "zoopark/index.html", context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\test\utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\test\utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\defaulttags.py", line 242, in render
    nodelist.append(node.render_annotated(context))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\templatetags\static.py", line 116, in render
    url = self.url(context)
          ^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\templatetags\static.py", line 112, in url
    path = self.path.resolve(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 710, in resolve
    obj = self.var.resolve(context)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 842, in resolve
    value = self._resolve_lookup(context)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 885, in _resolve_lookup
    current = getattr(current, bit)
              ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\fields\files.py", line 66, in url
    self._require_file()
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\fields\files.py", line 41, in _require_file
    raise ValueError(
ValueError: The 'image' attribute has no file associated with it.
2024-05-17 02:58:57,655 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\models.py changed, reloading.
2024-05-17 02:58:58,819 INFO Watching for file changes with StatReloader
2024-05-17 03:00:09,451 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\ZooparkApp\settings.py changed, reloading.
2024-05-17 03:00:10,634 INFO Watching for file changes with StatReloader
2024-05-17 03:00:29,353 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\models.py changed, reloading.
2024-05-17 03:00:30,431 INFO Watching for file changes with StatReloader
2024-05-17 03:01:43,526 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\models.py changed, reloading.
2024-05-17 03:01:44,678 INFO Watching for file changes with StatReloader
2024-05-17 03:01:57,407 ERROR Internal Server Error: /
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 875, in _resolve_lookup
    current = current[bit]
              ~~~~~~~^^^^^
TypeError: 'ImageFieldFile' object is not subscriptable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 21, in index
    return render(request, "zoopark/index.html", context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\test\utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\test\utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\defaulttags.py", line 242, in render
    nodelist.append(node.render_annotated(context))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\templatetags\static.py", line 116, in render
    url = self.url(context)
          ^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\templatetags\static.py", line 112, in url
    path = self.path.resolve(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 710, in resolve
    obj = self.var.resolve(context)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 842, in resolve
    value = self._resolve_lookup(context)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 885, in _resolve_lookup
    current = getattr(current, bit)
              ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\fields\files.py", line 66, in url
    self._require_file()
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\fields\files.py", line 41, in _require_file
    raise ValueError(
ValueError: The 'image' attribute has no file associated with it.
2024-05-17 03:02:57,464 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\models.py changed, reloading.
2024-05-17 03:02:58,716 INFO Watching for file changes with StatReloader
2024-05-17 03:05:37,585 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\ZooparkApp\settings.py changed, reloading.
2024-05-17 03:05:38,746 INFO Watching for file changes with StatReloader
2024-05-17 03:06:47,240 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\models.py changed, reloading.
2024-05-17 03:06:48,429 INFO Watching for file changes with StatReloader
2024-05-17 03:07:32,132 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\models.py changed, reloading.
2024-05-17 03:07:33,308 INFO Watching for file changes with StatReloader
2024-05-17 03:09:26,798 INFO Имя: gdfgs Вид: 2 Семейство: 4 добавлен в зоопарк.
2024-05-17 03:09:28,529 ERROR Internal Server Error: /
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 875, in _resolve_lookup
    current = current[bit]
              ~~~~~~~^^^^^
TypeError: 'ImageFieldFile' object is not subscriptable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 21, in index
    return render(request, "zoopark/index.html", context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\test\utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\test\utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\defaulttags.py", line 242, in render
    nodelist.append(node.render_annotated(context))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\templatetags\static.py", line 116, in render
    url = self.url(context)
          ^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\templatetags\static.py", line 112, in url
    path = self.path.resolve(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 710, in resolve
    obj = self.var.resolve(context)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 842, in resolve
    value = self._resolve_lookup(context)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 885, in _resolve_lookup
    current = getattr(current, bit)
              ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\fields\files.py", line 66, in url
    self._require_file()
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\fields\files.py", line 41, in _require_file
    raise ValueError(
ValueError: The 'image' attribute has no file associated with it.
2024-05-17 03:09:46,343 ERROR Internal Server Error: /management/
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 875, in _resolve_lookup
    current = current[bit]
              ~~~~~~~^^^^^
TypeError: 'ImageFieldFile' object is not subscriptable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 126, in management
    return render(request, "zoopark/management.html", context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\test\utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\test\utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\defaulttags.py", line 242, in render
    nodelist.append(node.render_annotated(context))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\templatetags\static.py", line 116, in render
    url = self.url(context)
          ^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\templatetags\static.py", line 112, in url
    path = self.path.resolve(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 710, in resolve
    obj = self.var.resolve(context)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 842, in resolve
    value = self._resolve_lookup(context)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 885, in _resolve_lookup
    current = getattr(current, bit)
              ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\fields\files.py", line 66, in url
    self._require_file()
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\fields\files.py", line 41, in _require_file
    raise ValueError(
ValueError: The 'image' attribute has no file associated with it.
2024-05-17 03:10:47,605 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\forms.py changed, reloading.
2024-05-17 03:10:49,093 INFO Watching for file changes with StatReloader
2024-05-17 03:10:55,495 INFO Имя: gdfgs Вид: 2 Семейство: 4 добавлен в зоопарк.
2024-05-17 03:10:57,675 ERROR Internal Server Error: /
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 875, in _resolve_lookup
    current = current[bit]
              ~~~~~~~^^^^^
TypeError: 'ImageFieldFile' object is not subscriptable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 21, in index
    return render(request, "zoopark/index.html", context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\test\utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\test\utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\defaulttags.py", line 242, in render
    nodelist.append(node.render_annotated(context))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\templatetags\static.py", line 116, in render
    url = self.url(context)
          ^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\templatetags\static.py", line 112, in url
    path = self.path.resolve(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 710, in resolve
    obj = self.var.resolve(context)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 842, in resolve
    value = self._resolve_lookup(context)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 885, in _resolve_lookup
    current = getattr(current, bit)
              ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\fields\files.py", line 66, in url
    self._require_file()
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\fields\files.py", line 41, in _require_file
    raise ValueError(
ValueError: The 'image' attribute has no file associated with it.
2024-05-17 03:11:45,307 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\models.py changed, reloading.
2024-05-17 03:11:46,767 INFO Watching for file changes with StatReloader
2024-05-17 03:12:04,127 INFO Имя: gdfgs Вид: 2 Семейство: 4 добавлен в зоопарк.
2024-05-17 03:12:08,194 INFO Имя: gdfgs Вид: 2 Семейство: 4 добавлен в зоопарк.
2024-05-17 03:12:10,611 ERROR Internal Server Error: /
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 875, in _resolve_lookup
    current = current[bit]
              ~~~~~~~^^^^^
TypeError: 'ImageFieldFile' object is not subscriptable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 21, in index
    return render(request, "zoopark/index.html", context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\test\utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\test\utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\defaulttags.py", line 242, in render
    nodelist.append(node.render_annotated(context))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\templatetags\static.py", line 116, in render
    url = self.url(context)
          ^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\templatetags\static.py", line 112, in url
    path = self.path.resolve(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 710, in resolve
    obj = self.var.resolve(context)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 842, in resolve
    value = self._resolve_lookup(context)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 885, in _resolve_lookup
    current = getattr(current, bit)
              ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\fields\files.py", line 66, in url
    self._require_file()
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\fields\files.py", line 41, in _require_file
    raise ValueError(
ValueError: The 'image' attribute has no file associated with it.
2024-05-17 03:13:19,818 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\forms.py changed, reloading.
2024-05-17 03:13:21,566 INFO Watching for file changes with StatReloader
2024-05-17 03:13:24,317 INFO Имя: gdfgs Вид: 2 Семейство: 4 добавлен в зоопарк.
2024-05-17 03:13:29,914 INFO Имя: gdfgs Вид: 2 Семейство: 4 добавлен в зоопарк.
2024-05-17 03:16:51,940 INFO Имя: fdgf Вид: 4 Семейство: 4 добавлен в зоопарк.
2024-05-17 03:17:37,956 ERROR Internal Server Error: /^animal/(?P21\d+)/$
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 875, in _resolve_lookup
    current = current[bit]
              ~~~~~~~^^^^^
TypeError: 'ImageFieldFile' object is not subscriptable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 60, in animal
    return render(request, "zoopark/animal.html", context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\test\utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\test\utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\templatetags\static.py", line 116, in render
    url = self.url(context)
          ^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\templatetags\static.py", line 112, in url
    path = self.path.resolve(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 710, in resolve
    obj = self.var.resolve(context)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 842, in resolve
    value = self._resolve_lookup(context)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 885, in _resolve_lookup
    current = getattr(current, bit)
              ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\fields\files.py", line 66, in url
    self._require_file()
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\fields\files.py", line 41, in _require_file
    raise ValueError(
ValueError: The 'image' attribute has no file associated with it.
2024-05-17 03:18:14,856 INFO Имя: fdgf Вид: Кенгуру Семейство: Кенгуровые удален из зоопарка.
2024-05-17 03:18:47,444 INFO Имя: fdgf Вид: 4 Семейство: 5 добавлен в зоопарк.
2024-05-17 03:19:06,150 ERROR Internal Server Error: /management/
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 875, in _resolve_lookup
    current = current[bit]
              ~~~~~~~^^^^^
TypeError: 'ImageFieldFile' object is not subscriptable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 126, in management
    return render(request, "zoopark/management.html", context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\test\utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\test\utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\defaulttags.py", line 242, in render
    nodelist.append(node.render_annotated(context))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\templatetags\static.py", line 116, in render
    url = self.url(context)
          ^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\templatetags\static.py", line 112, in url
    path = self.path.resolve(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 710, in resolve
    obj = self.var.resolve(context)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 842, in resolve
    value = self._resolve_lookup(context)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 885, in _resolve_lookup
    current = getattr(current, bit)
              ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\fields\files.py", line 66, in url
    self._require_file()
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\fields\files.py", line 41, in _require_file
    raise ValueError(
ValueError: The 'image' attribute has no file associated with it.
2024-05-17 03:19:29,215 ERROR Internal Server Error: /statistic/
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 875, in _resolve_lookup
    current = current[bit]
              ~~~~~~~^^^^^
TypeError: 'ImageFieldFile' object is not subscriptable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 108, in statistic
    return render(request, "zoopark/statistic.html", context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\test\utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\test\utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\defaulttags.py", line 242, in render
    nodelist.append(node.render_annotated(context))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\templatetags\static.py", line 116, in render
    url = self.url(context)
          ^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\templatetags\static.py", line 112, in url
    path = self.path.resolve(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 710, in resolve
    obj = self.var.resolve(context)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 842, in resolve
    value = self._resolve_lookup(context)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 885, in _resolve_lookup
    current = getattr(current, bit)
              ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\fields\files.py", line 66, in url
    self._require_file()
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\fields\files.py", line 41, in _require_file
    raise ValueError(
ValueError: The 'image' attribute has no file associated with it.
2024-05-17 03:21:06,952 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 03:21:10,027 INFO Frick Ronald вошел в аккаунт.
2024-05-17 03:21:11,128 ERROR Internal Server Error: /auth/profile/
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 875, in _resolve_lookup
    current = current[bit]
              ~~~~~~~^^^^^
TypeError: 'ImageFieldFile' object is not subscriptable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py", line 27, in profile
    return render(request, "users/profile.html", context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\test\utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\test\utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\defaulttags.py", line 326, in render
    return nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\defaulttags.py", line 242, in render
    nodelist.append(node.render_annotated(context))
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\templatetags\static.py", line 116, in render
    url = self.url(context)
          ^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\templatetags\static.py", line 112, in url
    path = self.path.resolve(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 710, in resolve
    obj = self.var.resolve(context)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 842, in resolve
    value = self._resolve_lookup(context)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 885, in _resolve_lookup
    current = getattr(current, bit)
              ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\fields\files.py", line 66, in url
    self._require_file()
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\db\models\fields\files.py", line 41, in _require_file
    raise ValueError(
ValueError: The 'image' attribute has no file associated with it.
2024-05-17 03:22:19,740 INFO Frick Ronald вышел из аккаунта.
2024-05-17 03:22:26,990 INFO Щур Марат вошел в аккаунт.
2024-05-17 03:22:31,677 INFO Щур Марат вышел из аккаунта.
2024-05-17 03:22:39,754 INFO London Jack вошел в аккаунт.
2024-05-17 03:22:59,527 INFO London Jack вышел из аккаунта.
2024-05-17 03:23:06,426 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 03:23:58,240 INFO Имя: fdgf Вид: Кенгуру Семейство: Ложноногие изменен.
2024-05-17 03:24:05,618 INFO Имя: fdgf Вид: Кенгуру Семейство: Ложноногие удален из зоопарка.
2024-05-17 03:27:37,507 INFO Имя: Клин Вид: 8 Семейство: 6 добавлен в зоопарк.
2024-05-17 03:30:35,348 INFO Получен случайный факт про кошек.
2024-05-17 03:30:37,114 INFO Получено случайное фото собаки.
2024-05-17 03:41:17,216 INFO Получено случайное фото собаки.
2024-05-17 03:41:19,312 INFO Получен случайный факт про кошек.
2024-05-17 03:47:25,052 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 03:47:46,929 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 03:47:48,425 INFO Watching for file changes with StatReloader
2024-05-17 03:47:51,388 INFO Frick Ronald вошел в аккаунт.
2024-05-17 03:47:57,226 WARNING Not Found: /accounts/login/
2024-05-17 03:48:04,852 INFO Frick Ronald вышел из аккаунта.
2024-05-17 03:48:11,592 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 03:48:30,084 WARNING Not Found: /accounts/login/
2024-05-17 03:49:12,341 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 03:49:13,333 INFO Watching for file changes with StatReloader
2024-05-17 03:49:25,201 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 03:51:41,977 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 03:51:43,138 INFO Watching for file changes with StatReloader
2024-05-17 03:51:46,081 WARNING Not Found: /accounts/login/
2024-05-17 03:56:03,378 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 03:56:04,343 INFO Watching for file changes with StatReloader
2024-05-17 03:56:29,188 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 03:56:30,384 INFO Watching for file changes with StatReloader
2024-05-17 03:56:32,212 WARNING Not Found: /accounts/login/
2024-05-17 03:56:43,060 INFO Frick Ronald вошел в аккаунт.
2024-05-17 03:56:46,248 WARNING Not Found: /accounts/login/
2024-05-17 03:57:00,194 WARNING Not Found: /registration/
2024-05-17 03:57:13,970 INFO Frick Ronald вышел из аккаунта.
2024-05-17 03:57:18,224 WARNING Not Found: /accounts/login/
2024-05-17 03:57:23,883 WARNING Not Found: /accounts/login/
2024-05-17 03:57:45,900 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 04:01:33,869 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 04:01:35,080 INFO Watching for file changes with StatReloader
2024-05-17 04:03:23,212 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 04:03:24,318 INFO Watching for file changes with StatReloader
2024-05-17 04:05:48,011 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 04:05:49,150 INFO Watching for file changes with StatReloader
2024-05-17 04:06:14,886 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 04:06:18,255 INFO Frick Ronald вошел в аккаунт.
2024-05-17 04:09:11,316 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 04:09:12,639 INFO Watching for file changes with StatReloader
2024-05-17 04:09:40,140 ERROR Internal Server Error: /auth/profile/
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 23, in _wrapper_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py", line 40, in profile
    context["user_tz"] = get_user_timezone(request)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py", line 47, in get_user_timezone
    return user.timezone
           ^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\utils\functional.py", line 253, in inner
    return func(_wrapped, *args)
           ^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'User' object has no attribute 'timezone'
2024-05-17 04:11:00,623 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 04:11:01,994 INFO Watching for file changes with StatReloader
2024-05-17 04:11:29,653 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 04:11:30,846 INFO Watching for file changes with StatReloader
2024-05-17 04:11:49,559 ERROR Internal Server Error: /auth/profile/
Traceback (most recent call last):
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py", line 48, in get_user_timezone
    user_tz = request.user.profile.timezone
              ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\utils\functional.py", line 253, in inner
    return func(_wrapped, *args)
           ^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'User' object has no attribute 'profile'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 23, in _wrapper_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py", line 41, in profile
    context["user_tz"] = get_user_timezone(request)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py", line 49, in get_user_timezone
    except (AttributeError, User.profile.RelatedObjectDoesNotExist):
                            ^^^^^^^^^^^^
AttributeError: type object 'User' has no attribute 'profile'
2024-05-17 04:13:10,939 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 04:13:12,274 INFO Watching for file changes with StatReloader
2024-05-17 04:14:36,472 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\ZooparkApp\settings.py changed, reloading.
2024-05-17 04:14:37,715 INFO Watching for file changes with StatReloader
2024-05-17 04:15:04,997 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\ZooparkApp\settings.py changed, reloading.
2024-05-17 04:15:06,185 INFO Watching for file changes with StatReloader
2024-05-17 04:15:15,932 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\ZooparkApp\settings.py changed, reloading.
2024-05-17 04:15:17,220 INFO Watching for file changes with StatReloader
2024-05-17 04:15:29,968 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\ZooparkApp\settings.py changed, reloading.
2024-05-17 04:15:31,130 INFO Watching for file changes with StatReloader
2024-05-17 04:17:27,321 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\ZooparkApp\settings.py changed, reloading.
2024-05-17 04:17:28,511 INFO Watching for file changes with StatReloader
2024-05-17 04:18:27,311 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\ZooparkApp\settings.py changed, reloading.
2024-05-17 05:18:28,510 INFO Watching for file changes with StatReloader
2024-05-17 05:18:41,161 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\ZooparkApp\settings.py changed, reloading.
2024-05-17 05:18:42,260 INFO Watching for file changes with StatReloader
2024-05-17 05:19:16,202 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\ZooparkApp\settings.py changed, reloading.
2024-05-17 05:19:17,250 INFO Watching for file changes with StatReloader
2024-05-17 05:22:00,328 INFO Watching for file changes with StatReloader
2024-05-17 05:31:33,342 INFO Watching for file changes with StatReloader
2024-05-17 05:31:49,576 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\ZooparkApp\settings.py changed, reloading.
2024-05-17 05:31:50,723 INFO Watching for file changes with StatReloader
2024-05-17 05:32:06,745 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\ZooparkApp\settings.py changed, reloading.
2024-05-17 05:32:07,873 INFO Watching for file changes with StatReloader
2024-05-17 05:34:33,569 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 04:34:34,825 INFO Watching for file changes with StatReloader
2024-05-17 04:36:12,577 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 04:36:13,695 INFO Watching for file changes with StatReloader
2024-05-17 04:43:40,476 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 04:43:41,618 INFO Watching for file changes with StatReloader
2024-05-17 04:44:17,733 ERROR Internal Server Error: /auth/profile/
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 23, in _wrapper_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py", line 43, in profile
    context["utc"] = timezone.now().astimezone(timezone.utc)
                                           ^^^^^^^^^^^^
AttributeError: module 'django.utils.timezone' has no attribute 'utc'
2024-05-17 04:45:26,647 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 04:45:27,765 INFO Watching for file changes with StatReloader
2024-05-17 04:47:27,328 ERROR Internal Server Error: /auth/profile/
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 23, in _wrapper_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py", line 44, in profile
    return render(request, "users/profile.html", context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\backends\django.py", line 61, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\test\utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader_tags.py", line 159, in render
    return compiled_parent._render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\test\utils.py", line 111, in instrumented_test_render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\loader_tags.py", line 65, in render
    result = block.nodelist.render(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1000, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 961, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 1059, in render
    output = self.filter_expression.resolve(context)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\base.py", line 737, in resolve
    new_obj = func(obj, *arg_vals)
              ^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\template\defaultfilters.py", line 787, in date
    return formats.date_format(value, arg)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\utils\formats.py", line 155, in date_format
    return dateformat.format(
           ^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\utils\dateformat.py", line 325, in format
    return df.format(format_string)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\utils\dateformat.py", line 45, in format
    raise TypeError(
TypeError: The format for date objects may not contain time-related format specifiers (found 'H').
2024-05-17 04:47:59,041 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 04:48:00,227 INFO Watching for file changes with StatReloader
2024-05-17 04:49:13,897 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 04:49:14,882 INFO Watching for file changes with StatReloader
2024-05-17 04:52:55,342 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 04:52:56,355 INFO Watching for file changes with StatReloader
2024-05-17 04:53:04,604 ERROR Internal Server Error: /auth/profile/
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 23, in _wrapper_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py", line 43, in profile
    context["utc"] = datetime.now(tz=pytz.all_timezones[0])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: tzinfo argument must be None or of a tzinfo subclass, not type 'str'
2024-05-17 04:54:01,778 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 04:54:02,794 INFO Watching for file changes with StatReloader
2024-05-17 04:54:24,499 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 04:54:25,703 INFO Watching for file changes with StatReloader
2024-05-17 04:55:35,286 INFO Watching for file changes with StatReloader
2024-05-17 04:55:53,116 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\ZooparkApp\settings.py changed, reloading.
2024-05-17 04:55:54,396 INFO Watching for file changes with StatReloader
2024-05-17 04:56:22,995 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 04:56:24,109 INFO Watching for file changes with StatReloader
2024-05-17 09:56:03,006 INFO Watching for file changes with StatReloader
2024-05-17 09:56:08,951 WARNING Not Found: /favicon.ico
2024-05-17 09:56:10,830 INFO Frick Ronald вышел из аккаунта.
2024-05-17 09:58:35,982 INFO Watching for file changes with StatReloader
2024-05-17 09:59:34,788 INFO User Test зарегистрировался на сайте.
2024-05-17 10:06:26,547 INFO Watching for file changes with StatReloader
2024-05-17 10:13:52,275 INFO Watching for file changes with StatReloader
2024-05-17 10:20:44,982 INFO Watching for file changes with StatReloader
2024-05-17 10:46:58,196 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 10:46:58,823 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 10:49:55,093 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 10:49:55,704 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 10:51:56,722 INFO Watching for file changes with StatReloader
2024-05-17 10:52:29,850 INFO User Test зарегистрировался на сайте.
2024-05-17 10:54:52,430 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 10:54:53,079 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 10:55:19,965 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 10:55:20,585 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 10:55:56,932 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 10:55:57,559 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 10:56:51,461 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 10:56:52,068 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 10:57:26,468 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 10:57:27,081 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 10:59:51,530 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 10:59:52,131 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 11:00:11,813 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 11:00:12,468 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 11:03:32,232 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 11:03:32,843 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 11:03:57,601 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 11:03:58,228 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 11:04:19,556 INFO Watching for file changes with StatReloader
2024-05-17 11:04:24,639 INFO Frick Ronald вошел в аккаунт.
2024-05-17 11:04:26,537 INFO Frick Ronald вышел из аккаунта.
2024-05-17 11:04:48,284 INFO sfdgfsdg gsdfgfds зарегистрировался на сайте.
2024-05-17 11:05:58,379 INFO sfdgfsdg gsdfgfds вошел в аккаунт.
2024-05-17 11:06:48,360 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 11:06:48,895 INFO Watching for file changes with StatReloader
2024-05-17 11:06:53,104 INFO sfdgfsdg gsdfgfds вышел из аккаунта.
2024-05-17 11:07:12,261 INFO fdsf rewrew зарегистрировался на сайте.
2024-05-17 11:07:21,065 INFO fdsf rewrew вышел из аккаунта.
2024-05-17 11:08:18,484 INFO Watching for file changes with StatReloader
2024-05-17 11:08:39,287 INFO gfsdgs sdfg зарегистрировался на сайте.
2024-05-17 11:08:55,518 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 11:08:56,176 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 11:09:23,616 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 11:09:24,239 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 11:09:57,074 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 11:09:57,697 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 11:12:24,854 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 11:12:25,524 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 11:12:41,661 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 11:12:42,298 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 11:13:50,187 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 11:13:50,858 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 11:17:08,019 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 11:17:08,684 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 11:21:11,657 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 11:21:12,279 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 11:23:21,908 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 11:23:22,538 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 11:23:46,354 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 11:23:46,980 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 11:24:27,162 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 11:24:27,775 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 11:26:03,047 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 11:26:03,687 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 11:26:35,991 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 11:26:36,609 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 11:27:18,080 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 11:27:18,692 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 12:48:56,145 INFO Watching for file changes with StatReloader
2024-05-17 12:49:10,957 ERROR Internal Server Error: /auth/profile/
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 23, in _wrapper_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py", line 44, in profile
    context["utc"] = last_change_datetime.now(tz=pytz.utc)
                 ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'now'
2024-05-17 12:49:21,941 INFO gfsdgs sdfg вышел из аккаунта.
2024-05-17 12:49:30,225 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 12:49:43,741 INFO Промокод Njd67H добавлен, скидка 6%.
2024-05-17 12:49:46,391 ERROR Internal Server Error: /auth/profile/
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 23, in _wrapper_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py", line 44, in profile
    context["utc"] = last_change_datetime.now(tz=pytz.utc)
                 ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'now'
2024-05-17 12:50:33,299 INFO Watching for file changes with StatReloader
2024-05-17 12:52:12,184 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 12:52:18,191 INFO Промокод MkJ78E добавлен, скидка 8%.
2024-05-17 12:52:18,572 ERROR Internal Server Error: /^add-promocode/$
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 23, in _wrapper_view
    return view_func(request, *args, **kwargs)
  File "c:\Users\drine\.vscode\extensions\ms-python.debugpy-2024.6.0-win32-x64\bundled\libs\debugpy\_vendored\pydevd\_pydevd_bundle\pydevd_trace_dispatch_regular.py", line 203, in trace_dispatch
    py_db.enable_tracing(thread_trace_func)
  File "c:\Users\drine\.vscode\extensions\ms-python.debugpy-2024.6.0-win32-x64\bundled\libs\debugpy\_vendored\pydevd\pydevd.py", line 1107, in enable_tracing
    pydevd_tracing.SetTrace(thread_trace_func)
  File "c:\Users\drine\.vscode\extensions\ms-python.debugpy-2024.6.0-win32-x64\bundled\libs\debugpy\_vendored\pydevd\pydevd_tracing.py", line 87, in SetTrace
    if set_trace_to_threads(tracing_func, thread_idents=[thread.get_ident()], create_dummy_thread=False) == 0:
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\drine\.vscode\extensions\ms-python.debugpy-2024.6.0-win32-x64\bundled\libs\debugpy\_vendored\pydevd\pydevd_tracing.py", line 354, in set_trace_to_threads
    start_new_thread(increase_tracing_count, ())
RuntimeError: can't create new thread at interpreter shutdown
2024-05-17 12:52:21,015 INFO Watching for file changes with StatReloader
2024-05-17 12:54:47,671 INFO Watching for file changes with StatReloader
2024-05-17 12:56:13,426 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 12:56:22,274 INFO Промокод n45Hn8 добавлен, скидка 3%.
2024-05-17 12:56:22,631 ERROR Internal Server Error: /^add-promocode/$
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 23, in _wrapper_view
    return view_func(request, *args, **kwargs)
  File "c:\Users\drine\.vscode\extensions\ms-python.debugpy-2024.6.0-win32-x64\bundled\libs\debugpy\_vendored\pydevd\_pydevd_bundle\pydevd_trace_dispatch_regular.py", line 203, in trace_dispatch
    py_db.enable_tracing(thread_trace_func)
  File "c:\Users\drine\.vscode\extensions\ms-python.debugpy-2024.6.0-win32-x64\bundled\libs\debugpy\_vendored\pydevd\pydevd.py", line 1107, in enable_tracing
    pydevd_tracing.SetTrace(thread_trace_func)
  File "c:\Users\drine\.vscode\extensions\ms-python.debugpy-2024.6.0-win32-x64\bundled\libs\debugpy\_vendored\pydevd\pydevd_tracing.py", line 87, in SetTrace
    if set_trace_to_threads(tracing_func, thread_idents=[thread.get_ident()], create_dummy_thread=False) == 0:
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\drine\.vscode\extensions\ms-python.debugpy-2024.6.0-win32-x64\bundled\libs\debugpy\_vendored\pydevd\pydevd_tracing.py", line 354, in set_trace_to_threads
    start_new_thread(increase_tracing_count, ())
RuntimeError: can't create new thread at interpreter shutdown
2024-05-17 12:56:33,543 INFO Watching for file changes with StatReloader
2024-05-17 12:57:09,752 INFO Промокод jhUj67 добавлен, скидка 4%.
2024-05-17 12:58:18,820 INFO Watching for file changes with StatReloader
2024-05-17 13:00:12,566 INFO Watching for file changes with StatReloader
2024-05-17 13:01:45,642 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py changed, reloading.
2024-05-17 13:01:46,692 INFO Watching for file changes with StatReloader
2024-05-17 13:02:14,779 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 13:02:15,886 INFO Watching for file changes with StatReloader
2024-05-17 13:03:19,943 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\last_time.py changed, reloading.
2024-05-17 13:03:21,062 INFO Watching for file changes with StatReloader
2024-05-17 13:03:28,446 ERROR Internal Server Error: /auth/profile/
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 23, in _wrapper_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py", line 45, in profile
    context["utc"] = last_change_datetime.now(tz=pytz.utc)
                 ^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'now'
2024-05-17 13:03:56,838 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\last_time.py changed, reloading.
2024-05-17 13:03:57,845 INFO Watching for file changes with StatReloader
2024-05-17 13:07:46,076 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\ZooparkApp\settings.py changed, reloading.
2024-05-17 13:07:47,199 INFO Watching for file changes with StatReloader
2024-05-17 13:09:44,266 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\ZooparkApp\settings.py changed, reloading.
2024-05-17 13:09:45,575 INFO Watching for file changes with StatReloader
2024-05-17 13:11:51,889 INFO Watching for file changes with StatReloader
2024-05-17 13:16:57,306 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\ZooparkApp\settings.py changed, reloading.
2024-05-17 13:16:58,493 INFO Watching for file changes with StatReloader
2024-05-17 13:18:29,802 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 13:18:30,930 INFO Watching for file changes with StatReloader
2024-05-17 13:18:33,069 ERROR Internal Server Error: /auth/profile/
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 23, in _wrapper_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py", line 45, in profile
    context["utc"] = timezone.now(tz=pytz.utc)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: now() got an unexpected keyword argument 'tz'
2024-05-17 13:19:18,176 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 13:19:19,239 INFO Watching for file changes with StatReloader
2024-05-17 13:20:24,368 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\ZooparkApp\settings.py changed, reloading.
2024-05-17 13:20:25,534 INFO Watching for file changes with StatReloader
2024-05-17 13:20:51,739 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\ZooparkApp\settings.py changed, reloading.
2024-05-17 13:20:52,960 INFO Watching for file changes with StatReloader
2024-05-17 13:21:33,832 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 13:21:34,973 INFO Watching for file changes with StatReloader
2024-05-17 13:22:33,556 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\ZooparkApp\settings.py changed, reloading.
2024-05-17 13:22:34,683 INFO Watching for file changes with StatReloader
2024-05-17 13:27:23,139 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\ZooparkApp\settings.py changed, reloading.
2024-05-17 13:27:24,300 INFO Watching for file changes with StatReloader
2024-05-17 13:31:18,181 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 13:31:19,289 INFO Watching for file changes with StatReloader
2024-05-17 13:32:19,995 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 13:32:21,186 INFO Watching for file changes with StatReloader
2024-05-17 13:32:50,324 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 13:32:51,495 INFO Watching for file changes with StatReloader
2024-05-17 13:33:09,131 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 13:33:10,307 INFO Watching for file changes with StatReloader
2024-05-17 13:34:03,535 INFO D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\users\views.py changed, reloading.
2024-05-17 13:34:04,765 INFO Watching for file changes with StatReloader
2024-05-17 13:34:11,505 INFO Фамилия Имя вышел из аккаунта.
2024-05-17 13:35:25,081 INFO Frick Ronald вошел в аккаунт.
2024-05-17 13:35:30,080 ERROR Internal Server Error: /^cat-fact/$
Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\urllib3\connection.py", line 198, in _new_conn
    sock = connection.create_connection(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\urllib3\util\connection.py", line 60, in create_connection
    for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\drine\AppData\Local\Programs\Python\Python312\Lib\socket.py", line 963, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
socket.gaierror: [Errno 11001] getaddrinfo failed

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\urllib3\connectionpool.py", line 793, in urlopen
    response = self._make_request(
               ^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\urllib3\connectionpool.py", line 491, in _make_request
    raise new_e
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\urllib3\connectionpool.py", line 467, in _make_request
    self._validate_conn(conn)
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\urllib3\connectionpool.py", line 1099, in _validate_conn
    conn.connect()
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\urllib3\connection.py", line 616, in connect
    self.sock = sock = self._new_conn()
                       ^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\urllib3\connection.py", line 205, in _new_conn
    raise NameResolutionError(self.host, self, e) from e
urllib3.exceptions.NameResolutionError: <urllib3.connection.HTTPSConnection object at 0x000002B605B66A80>: Failed to resolve 'catfact.ninja' ([Errno 11001] getaddrinfo failed)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\requests\adapters.py", line 486, in send
    resp = conn.urlopen(
           ^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\urllib3\connectionpool.py", line 847, in urlopen
    retries = retries.increment(
              ^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\urllib3\util\retry.py", line 515, in increment
    raise MaxRetryError(_pool, url, reason) from reason  # type: ignore[arg-type]
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='catfact.ninja', port=443): Max retries exceeded with url: /fact (Caused by NameResolutionError("<urllib3.connection.HTTPSConnection object at 0x000002B605B66A80>: Failed to resolve 'catfact.ninja' ([Errno 11001] getaddrinfo failed)"))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\django\contrib\auth\decorators.py", line 23, in _wrapper_view
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\views.py", line 235, in cat_fact
    "fact" : get_fact()}
             ^^^^^^^^^^
  File "D:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\zoopark\services\cat_facts.py", line 9, in get_fact
    response = requests.get(url)
               ^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\requests\api.py", line 73, in get
    return request("get", url, params=params, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\requests\api.py", line 59, in request
    return session.request(method=method, url=url, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\requests\sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\requests\sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "d:\Study\4 semestr\IGI\253502_Drinevskiy_8\IGI\LR5\ZooparkApp\.venv\Lib\site-packages\requests\adapters.py", line 519, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='catfact.ninja', port=443): Max retries exceeded with url: /fact (Caused by NameResolutionError("<urllib3.connection.HTTPSConnection object at 0x000002B605B66A80>: Failed to resolve 'catfact.ninja' ([Errno 11001] getaddrinfo failed)"))
2024-05-17 13:36:09,913 INFO Frick Ronald вышел из аккаунта.
2024-05-17 13:36:24,206 INFO Щур Марат вошел в аккаунт.
2024-05-17 13:36:39,425 INFO Щур Марат вышел из аккаунта.
2024-05-17 13:36:59,872 INFO Redkins Bill вошел в аккаунт.
2024-05-17 13:37:39,598 INFO Redkins Bill купил билет за 42.77 рублей.
2024-05-17 13:37:53,549 INFO Получен случайный факт про кошек.
2024-05-17 13:38:02,591 INFO Получено случайное фото собаки.
2024-05-17 13:38:05,333 INFO Redkins Bill вышел из аккаунта.
2024-05-17 13:38:13,310 INFO Фамилия Имя вошел в аккаунт.
2024-05-17 18:55:16,549 INFO Watching for file changes with StatReloader
2024-05-17 18:55:20,466 WARNING Not Found: /favicon.ico
